cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -78,9 +85,24 @@ public DatasetSpecification create(Id.DatasetInstance datasetInstanceId, Dataset
       }
 
       DatasetSpecification spec = type.configure(datasetInstanceId.getId(), props);
-      DatasetAdmin admin = type.getAdmin(DatasetContext.from(datasetInstanceId.getNamespaceId()), spec);
+      DatasetContext context = DatasetContext.from(datasetInstanceId.getNamespaceId());
+      DatasetAdmin admin = type.getAdmin(context, spec);
       try {
         admin.create();
+
+        // add system metadata for user datasets only
+        if (isUserDataset(datasetInstanceId)) {
+          try (Dataset dataset = type.getDataset(context, spec, DatasetDefinition.NO_ARGUMENTS)) {
```

Comment:
Is it guaranteed that a dataset can be instantiated without runtime arguments? I am not aware of one that cannot, but I don't believe there is a contract that prohibits a dataset from requiring arguments. So this might throw an exception, in which case we should still be able to record the metadata, just we can't derive BATCH or EXPLORE